rakan/cRakan/src/Runner.h

```diff
@@ -24,68 +24,96 @@ class Runner {
-
+  /**
+   * Destructor.
+   */
+  ~Runner();
+
+  /**
+   * Adds a new node to the graph this Runner instance runs on.
+   * 
+   * @param   node_id               the node id of the node to add
+   * @param   county                the county the to be added node resides in
+   * @param   majority_population   the majority population in the node
+   * @param   minority_population   the minority population in the node
+   */
+  void add_node(uint32_t node_id,
+                uint32_t county,
+                uint32_t majority_population,
+                uint32_t minority_population);
+
+  /**
+   * Adds an edge from node_one to node_two and from node_two to node_one
+   * on the graph this Runner runs on.
+   * 
+   * @param       node_one      the node to add an edge from node_two to
+   * @param       node_two      the node to add an edge from node_one to
+   */
   void add_edge(uint32_t node_one, uint32_t node_two);
 
-  // Returns a vector of vectors of uint32_t, where each vector
-  // represents the changes made to the graph in each step of 
-  // the walk.
-  vector<vector<uint32_t>&>& getMaps();
-  /*
-  * Sets the district assignments according to the given vector.
-  * Vector is interpreted as the node_id corresponding to the index
-  * of the district number inside of the vector.
-  * 
-  * @param    districts     a vector of uint32_t where each index of
-  *                         the vector corresponds to the node id.
+  /**
+  * Sets the district assignments according to the given vector. Vector is
+  * interpreted as the node_id corresponding to the index of the district
+  * number inside of the vector.
   * 
+  * @param    districts     a vector of uint32_t where each index of the vector
+  *                         corresponds to the node id
   */
   void set_districts(vector<uint32_t>& districts);
 
-  /*
-  * Generates random seeds on the current graph. Randomly selects
-  * a number of nodes to be the "center" of each district and assigns
-  * other nodes reachable from the seed nodes to the respective
-  * district.
+  /**
+  * Generates random seeds on the current graph. Randomly selects a number of
+  * nodes to be the "center" of each district and assigns other nodes reachable
+  * from the seed nodes to the respective district.
   * 
+  * @return true iff seeding is successful (i.e, all precincts are reachable),
+  *         fase otherwise
   */
   bool seed();
 
-  /*
-  * Populates the graph's data structures.
+  /**
+  * Populates the graph's data structures. Prerequisites to this method depend
+  * on the job.
+  * 
+  * If job is to generate a random seeding, walk on the randomly seeded graph,
+  * and return the score of the graph, `add_node()`, `add_edge`, and `seed()`
+  * must be called beforehand.
   * 
+  * If job is to return the score of the graph without random seeding, then
+  * `add_node()`, `add_edge()`, and `set_districts()` must be called
+  * beforehand.
   */
   void populate();
```

Comment:
Is it possible for this method to fail based on incomplete calls? Should this return true/false if it succeeds/fails?

Reply:
Unfortunately, there is no way for us to know if the necessary calls have been made. All there is to the graph is a bunch of unsigned numbers in fixed length arrays and that's all `populate()` sees. Should we note that it is up to the caller?

Reply:
nope, that's fine.